### PR TITLE
Linux+Proton support

### LINKED ISSUE #2: Add Linux+Proton support
Hey @biggestcookie, I wanted to adjust mouse sensitivity for TPS while playing under Linux+Proton (not the Linux native port), so I made a hardcoded dirty hack of your program for personal use.

I thought it was worth expanding my hack out to properly support players on Linux+Proton, [so I made a proof of concept fork](https://github.com/JonLiuFYI/Borderlands-Sensitivity-Changer) that mainly changes `profile.py` to support finding `profile.bin` on Linux+Proton. (This PoC doesn't support the Linux native ports of the games, only the Windows versions running on Linux using Proton.)

Are you interested in adding Linux support? If so, please check out my proof of concept. If you want this, I can start a pull request and continue working on it. The main work is cleaning up my code to be in line with yours and abstracting away OS-specific routines.

### PR DESCRIPTION
**Pull request by JonLiuFYI**
When complete, this closes #2 and increases the version number to 1.2.

## Tasks
- [x] Abstract away Windows/Linux differences from `profiles.py`
- [x] Conditionally import ctypes windll
- [x] Change Windows-specific hardcoded paths (like config.json) to work on linux
- [x] Update README: python min ver (3.8?), OS support
- [x] Authors file

## Acceptance
- [x] Works on Windows without regressions
- [x] Reproduce that it works on Linux+Proton
- [x] Works with PyInstaller

## After merging
Build and release version 1.2 for
- [ ] Windows
- [ ] Linux

### CODE REVIEW

**Comment by JonLiuFYI** [comment]
My changes to complete #2 are done. I live tested these changes on my PC running Fedora, and it properly changes mouse sensitivity and saves a config.json file.

46d9b1f changes how the `config_path` is determined, which allows PyInstaller onefiles built for Linux to find the right path.

**Comment by biggestcookie** [comment]
Thanks @JonLiuFYI -- will be giving it a look this week!

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# Borderlands Sensitivity Changer
+# Borderlands Sensitivity Changer 1.2
 
-### [Download it here!](https://github.com/biggestcookie/borderlands-sensitivity-changer/releases/latest)
+### [Download it here! (for Windows and Linux+Proton)](https://github.com/biggestcookie/borderlands-sensitivity-changer/releases/latest)
```
Comment (biggestcookie) [inline README.md]: Another possible wording change?
```suggestion
### [Download for Windows and Linux+Proton here!](https://github.com/biggestcookie/borderlands-sensitivity-changer/releases/latest)
```

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-# Borderlands Sensitivity Changer
+# Borderlands Sensitivity Changer 1.2
```
Comment (biggestcookie) [inline README.md]: I prefer the title in the README without the version number but that may just be me
```suggestion
# Borderlands Sensitivity Changer
```

**Review by biggestcookie — COMMENTED** [review]
Finished reviewing and testing the solution. Working great on my Windows machine. Once again really appreciate the stellar work and refactoring! I learned a lot such as Python having built-in abstract base class utilities.
Only have minor suggestions for the README that I do not mind much either way on, so after you review I'll approve, merge, and create a release here as well as on [Nexus Mods](https://www.nexusmods.com/borderlands2/mods/256).

File: README.md [hunk 3]
```diff
@@ -58,7 +58,7 @@ The 'Uncapped Pause Menu Settings' mod for [Borderlands 2](https://www.nexusmods
 
 ### Requirements
 
-[Python 3.5+](https://www.python.org/downloads/)
+[Python 3.8+](https://www.python.org/downloads/)
```
Comment (JonLiuFYI) [inline README.md]:
```suggestion
[Python 3.10+](https://www.python.org/downloads/)
```
I misremembered Python version numbers. It's 3.10 that introduced the typing syntax `Foo | Bar` as a replacement for `typing.Union[Foo, Bar]`. I used the new style in this PR.

**Comment by JonLiuFYI** [comment]
Ship this bad boy.

**Review by biggestcookie — APPROVED** [review]

**Pull request merged**